File: common/src/test/java/dev/number6/generate/PresentableEntityResultsGenerator.java
package dev.number6.generate;


import dev.number6.comprehend.results.PresentableEntityResults;
import uk.org.fyodor.generators.Generator;

import java.time.LocalDate;
import java.util.Map;

public class PresentableEntityResultsGenerator implements Generator<PresentableEntityResults> {

    Generator<Map<String, Map<String, Long>>> entityResultsGenerator = CommonRDG.map(CommonRDG.string(10),
            CommonRDG.map(CommonRDG.string(10), CommonRDG.longVal(100)));

    @Override
    public PresentableEntityResults next() {
        return new PresentableEntityResults(LocalDate.now(), entityResultsGenerator.next(), CommonRDG.string().next());
    }
}

File: common/src/test/java/dev/number6/generate/PresentableEntityResultsGenerator.kt
package dev.number6.generate

import dev.number6.comprehend.results.PresentableEntityResults
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG
import java.time.LocalDate

class PresentableEntityResultsGenerator : Generator<PresentableEntityResults> {
    var entityResultsGenerator = RDG.map(RDG.string(10),
            RDG.map(RDG.string(10), RDG.longVal(100)))

    override fun next(): PresentableEntityResults {
        return PresentableEntityResults(LocalDate.now(), entityResultsGenerator.next(), RDG.string().next())
    }
}